db/migrate/20210119101158_create_payment_intents.rb

```diff
@@ -0,0 +1,16 @@
+class CreatePaymentIntents < ActiveRecord::Migration[5.0]
+  disable_ddl_transaction!
+
+  def change
+    create_table :payment_intents do |t|
+      t.references :invoice, null: false
+      t.string :payment_intent_id
```

Comment:
shouldn't we enforce uniqueness? that would also help for the indexation. Like:
```ruby
t.string :payment_intent_id, unique: true, null: false
```
and below when you add the index:
```ruby
add_index :payment_intents, :payment_intent_id, index_options.merge({unique: true})
```